src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1004,6 +1009,44 @@ private void AppendDeadSegment(TimeSeriesValuesSegment newValueSegment)
                 EnsureStatsAndDataIntegrity(_context, _docId, _name, newValueSegment, _collection, BaselineDate);
             }
 
+            public void AppendToNewSegment(TimeSeriesValuesSegment newSegment, DateTime baseline, bool conflictStatusIsUpdate = false, string changeVector = null)
```

Comment:
let's avoid duplicate code, `AppendToNewSegment` should call this one, right?

Reply:
done